lib/termination-poller.js

```diff
@@ -0,0 +1,140 @@
+const assert = require('assert');
+const {runAWSRequest} = require('./aws-request');
+const Iterate = require('taskcluster-lib-iterate');
+const log = require('./log');
+
+const INSTANCE_LIMIT = 200;
+const POLL_DELAY = 100; // I forget which units this is :/
+
+class TerminationPoller {
+  constructor({ec2, regions, runaws = runAWSRequest, pollDelay = POLL_DELAY, state}) {
+    assert(typeof ec2 === 'object');
+    assert(Array.isArray(regions));
+    regions.forEach(region => assert(typeof region === 'string'));
+    assert(typeof state === 'object');
+
+    this.ec2 = ec2;
+    this.regions = regions;
+    this.state = state;
+    this.runaws = runaws;
+
+    this.iterator = new Iterate({
+      maxIterationTime: 1000 * 60 * 15,
+      watchDog: 1000 * 60 * 15,
+      maxFailures: 10,
+      waitTime: pollDelay,
+      handler: async(watchdog) => {
+        try {
+          await this.poll(watchdog);
+        } catch (err) {
+          console.dir(err.stack || err);
+          monitor.reportError(err, 'Error polling pricing data');
+        }
+      },
+    });
+  }
+
+  start() {
+    return this.iterator.start();
+  }
+
+  stop() {
+    return this.iterator.stop();
+  }
+
+  async poll(watchdog, sharedState) {
+    let toPoll = await this.state.findTerminationsToPoll(INSTANCE_LIMIT);
+
+    if (toPoll.length && toPoll.length === 0) {
+      return;
+    }
+
+    let resolutions = [];
+
+    await Promise.all(this.regions.map(async region => {
+
+      // Let's find the instance ids for our region
+      let instanceIds = toPoll.filter(x => x.region === region).map(x => x.id);
+      log.info({instanceIds, region}, 'polling terminations');
+
+      // We don't want to bother with the rest of this function if there's no
+      // instance ids in this region
+      //if (instanceIds.length === 0) {
+        //return;
+      //}
```

Comment:
It's usually best to avoid checking in commented-out code.  Either remove it, or if it was unintentional, un-comment it.